src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IDesignTimeCodeGenerator.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+internal interface IDesignTimeCodeGenerator
```

Comment:
IMO, forcing design-time code generation really doesn't belong as part of the `IDocumentSnapshot` contract for a couple of reasons:

1. The ability to force generation of a design-time code document was originally added as a temporary concession while we work through rolling out FUSE. Since it's temporary, making it easy for _all_ features to use it isn't really a goal. Instead, it's better to make features that need design-time code generation do a little extra work to get it.
2. Generating a design-time `RazorCodeDocument` doesn't cache the result. So, calling this has a performance cost and should be done sparingly, and only by the features that really need it.

Reply:
You make a very good argument